### PR TITLE
Add visible labels to BlockPatternPicker pattern selection buttons

### LINKED ISSUE #18737: Add visible labels to InnerBlocksTemplatePicker pattern selection buttons
**Is your feature request related to a problem? Please describe.**
The lack of visual labels within the `InnerBlocksTemplatePicker` component makes it difficult to understand what the `IconButton` stands for, i.e. what pattern will be added. Currently, the only way to discover the label is to wait for the tooltip.

### **Examples:**

**1. Core Columns block:** 
The lack of labels works _ok_ here, as the icons depict exactly what pattern will be added. Apart from this particular scenario, I can't think of many instances where these icons will depict the actual pattern. 
Screen Shot 2019-11-25 at 7 34 31 PM

**CoBlocks Form block:** 
It's not possible to include actual interpretations of these patterns within the CoBlocks Form block. The implementation is awesome, but the lack of labels make it difficult to understand exactly what pattern will be created - unless of course you wait for the tooltip label.
Screen Shot 2019-11-25 at 7 35 24 PM

**Describe the solution you'd like**
Either there's a prop to pass visible labels, or visible labels are always present. We'd want to consider if we should use the existing `label` prop, or if we should have separate `label` and `tooltip` props.

**Describe alternatives you've considered**
Recreating the component, though that's not a consideration really.

### PR DESCRIPTION
**Pull request by gziolo**
Fixes #18737.

## Description

> The lack of visual labels within the `BlockPatternPicker ` component makes it difficult to understand what the `IconButton` stands for, i.e. what pattern will be added. Currently, the only way to discover the label is to wait for the tooltip.

> **Describe the solution you'd like**
> Either there's a prop to pass visible labels, or visible labels are always present. We'd want to consider if we should use the existing `label` prop, or if we should have separate `label` and `tooltip` props.

I'm testing reusing the existing label of the pattern as the actual label rendered. In that case, we probably would have to improve copies for the columns block.

> Here's a quick mockup for what it could look like for the Core Columns block and the CoBlocks Form block:
>
> image

## Screenshots 

It needs more love on the CSS side of things:

Screen Shot 2020-01-21 at 18 49 12

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

**Comment by gziolo** [comment]
I would appreciate some help from designers. I have trouble doing it myself, mostly because of the fact that in the mock-ups the label is outside of the button. Ideally, both the icon and label should be wrapped with the Button component.

**Comment by ItsJonQ** [comment]
Haiii @gziolo !

After playing around with it for a while, I think the better UI may be to provide shorter labels underneath the buttons (rather than the long labels):

Screen Shot 2020-02-21 at 4 10 49 PM

To ensure screen readers have context, I've added the long labels as `aria-label` (tested with Mac OS Voiceover)

I think making the label clickable would be ideal. However, I think it may be quite involved to restyle the wrapper button, and simulate a fake button within (to allow for both Button + Label to be clickable).

Alternatively! I think perhaps these may be better if they were `input[type="radio"]`, as that is the UI we actually have :). Granted, we'd still have to do styling to resolve the Button/Label situation.

(Totally can be done later)

Anywhoo! Let me know if my pushed fixes are okay!

**Comment by mapk** [comment]
Looks good! It'll need a rebase with G2, but worked well regardless.

Thanks for jumping into this @ItsJonQ.

Screen Shot 2020-03-06 at 2 09 29 PM

**Comment by gziolo** [comment]
I rebased this branch with `master` and applied a few minor tweaks:

Screen Shot 2020-03-12 at 18 30 11

**Comment by gziolo** [comment]
@ItsJonQ thank you for all your help. I like the approach you took 👍

File: packages/block-editor/src/components/block-variation-picker/index.js [hunk 1]
```diff
@@ -46,8 +47,14 @@ function BlockVariationPicker( {
 							iconSize={ 48 }
 							onClick={ () => onSelect( variation ) }
 							className="block-editor-block-variation-picker__variation"
-							label={ variation.title }
+							label={ variation.description || variation.title }
```
Comment (gziolo) [inline packages/block-editor/src/components/block-variation-picker/index.js]: `description` is optional, so we should fallback to `title` when not provided.

**Review by youknowriad — COMMENTED** [review]
I didn't test but code-wise, this looks good to me. I'll leave the final call for designers.

**Comment by gziolo** [comment]
Let's process with this PR as is, we can always iterate later.

**Pull request merged**